specutils/analysis/flux.py

```diff
@@ -121,6 +129,15 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None):
 
     line_flux = _compute_line_flux(spectrum, regions)
 
+    # Account for the existence of a mask. A correction factor
+    # is applied to the continuum value to account for the
+    # data points that are not included in the line flux due to
+    # masking.
+    if hasattr(spectrum, 'mask') and spectrum.mask is not None:
+        valid = np.sum(np.where(calc_spectrum.mask, 0., 1.))
+        factor = valid / len(spectral_axis)
+        continuum *= factor
```

Comment:
I'm confused why this is necessary if we don't actually truncate the line flux calculation (that is, we instead replace with zeros).

Reply:
This is not very clear to me either. My idea was that, if we remove data points from the line flux summation (by setting their fluxes to zero), the continuuum value should be corrected by a factor that reflects the flux that was removed. If we don't do that, we can get into situations where the EW of an emission line, say, can even become positive. Again, I am not sure how to apply this correction, if indeed a correction exists or is necessary.

Reply:
@eteq can you expand on the necessity of this?